lib/handsoap/service.rb

```diff
@@ -265,7 +274,7 @@ def dispatch(doc, action)
         fire_on_log_header { "HandSoap Response [#{dispatch_id}]: length: [#{@http_client.body_str.length}], HTTP-Status: [#{@http_client.response_code}], Content-Type: [#{@http_client.content_type}]" }
         fire_on_log_body { Handsoap.pretty_format_envelope(@http_client.body_str) }
 
-        soap_response = Response.new(@http_client.body_str, envelope_namespace)
+        soap_response = Response.new(@http_client, envelope_namespace)
```

Comment:
What I was getting at was that you can only tell if serialization is needed by examining how `curb` maintains these values. If it maintains per-thread values, this is not needed. However, since this does seem to make the code a bit cleaner, it's still probably worthwhile.